lib/utils.pm

```diff
@@ -808,6 +809,22 @@ sub handle_login {
     send_key "ret";
 }
 
+sub handle_logout {
+    # hide mouse for clean logout needles
+    mouse_hide();
+    # logout
+    if (check_var('DESKTOP', 'gnome') || check_var('DESKTOP', 'lxde')) {
+        my $command = check_var('DESKTOP', 'gnome') ? 'gnome-session-quit' : 'lxsession-logout';
+        x11_start_program("$command");    # opens logout dialog
+        assert_screen 'logoutdialog' unless check_var('DESKTOP', 'gnome');
+    }
```

Comment:
whole block is over complicated IMO - I prefer 
if(check_var('DESKTOP', 'gnome')){
   x11_start_program('gnome-session-quit');   
}
elsif (check_var('DESKTOP', 'lxde')){
  x11_start_program('lxsession-logout');
  assert_screen 'logoutdialog';
}

which is less cool and optimal but more readable